pkg/apis/wksprovider/machine/os/os.go

```diff
@@ -814,73 +815,82 @@ func createAuthConfigMapManifest(authnSecretName, authzSecretName string, authnC
 // Decrypts secret, adds plan resources to install files found inside, plus a kubeconfig file pointing to them.
 // returns the sealed file contents, decrypted contents, secret name, kubeconfig, error if any
 func processSecret(b *plan.Builder, key *rsa.PrivateKey, configDir, secretFileName, URL string) ([]byte, map[string][]byte, string, []byte, error) {
+	// Read the file contents at configDir/secretFileName
 	contents, err := getConfigFileContents(configDir, secretFileName)
 	if err != nil {
 		return nil, nil, "", nil, err
 	}
-	object, err := runtime.Decode(scheme.Codecs.UniversalDecoder(ssv1alpha1.SchemeGroupVersion), contents)
-	if err != nil {
-		return nil, nil, "", nil, err
+
+	// Create a new YAML FrameReader from the given bytes
+	fr := serializer.NewYAMLFrameReader(serializer.FromBytes(contents))
+	// Create the secret to decode into
+	ss := &ssv1alpha1.SealedSecret{}
+	// Decode the Sealed Secret into the object
+	// In the future, if we wish to support other kinds of secrets than SealedSecrets, we
+	// can just change this to do .Decode(fr), and switch on the type
+	if err := scheme.Serializer.Decoder().DecodeInto(fr, ss); err != nil {
+		return nil, nil, "", nil, errors.Wrapf(err, "File %q does not contain a sealed secret, couldn't decode", secretFileName)
```

Comment:
Is it possible this error is something simpler, like bad indentation?

Reply:
Yes, changed the error message to be more generic.